src/Compilers/VisualBasic/Portable/Scanner/ScannerBuffer.vb

```diff
@@ -134,8 +134,78 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
             Return ch
         End Function
 
+        Friend Function TryPeek(ByRef c As Char) As Boolean
+            If _lineBufferOffset >= _bufferLen Then Return False
+            Dim page = _curPage
+            Dim pos = _lineBufferOffset
+            Dim ch = page._arr(pos And PAGE_MASK)
+            Dim start = page._pageStart
+            Dim expectedStart = pos And NOT_PAGE_MASK
+            If start <> expectedStart Then
+                page = GetPage(pos)
+                ch = page._arr(pos And PAGE_MASK)
+            End If
+            c = ch
+            Return True
+        End Function
+
+        Friend Function TryGet(ByRef c As String) As Boolean
+            If _lineBufferOffset >= _bufferLen Then Return False
+            Dim page = _curPage
+            Dim pos = _lineBufferOffset
+            Dim ch = page._arr(pos And PAGE_MASK)
+            Dim start = page._pageStart
+            Dim expectedStart = pos And NOT_PAGE_MASK
+
+            If start <> expectedStart Then
+                page = GetPage(pos)
+                ch = page._arr(pos And PAGE_MASK)
+            End If
+            c = Intern(ch)
+            Return True
+        End Function
+
+        ' PERF CRITICAL
+        Private Function TryPeek(skip As Integer, ByRef c As Char) As Boolean
+            ' CanGetCharAtOffset( )
+            If _lineBufferOffset + skip >= _bufferLen Then Return False
+            If skip < -MaxCharsLookBehind Then Return False
+            Dim pos = _lineBufferOffset
+            Dim page = _curPage
+            pos += skip
+            Dim ch = page._arr(pos And PAGE_MASK)
+            Dim start = page._pageStart
+            Dim expectedStart = pos And NOT_PAGE_MASK
+            If start <> expectedStart Then
+                page = GetPage(pos)
+                ch = page._arr(pos And PAGE_MASK)
+            End If
+            c = ch
+            Return True
+        End Function
+
+        Private Function IsPeek(at As Integer, isChar As Char) As Boolean
+            Dim c As Char
+            Return TryPeek(at, c) AndAlso (c = isChar)
+        End Function
+
+        Private Function ArePeek(chars As String) As Boolean
+            Return ArePeek(_lineBufferOffset, chars.Length, 1, chars)
```

Comment:
@pharring It "looks wrong" because it is peeking at characters offset from the `at` position.
Think of `_lineBufferOffset` as where we are in the source text. 
If you examine the internal of it's overload.
```
 For i = skip To (size - 1)
     If Peek(at + i) <> chars(i - skip) Then Return False
 Next
```

```
source  chars{ ]
(at+1)   +0
(at+2)   +1
(at+3)   +2
(at+4)   +3
```
The overload handles most of the case where `skip` isn't 1. Some I didn't bother with as I thought that the existing code would be better.